x-pack/plugin/eql/src/main/antlr/EqlBase.g4

```diff
@@ -212,6 +212,7 @@ LP: '(';
 RP: ')';
 PIPE: '|';
 OPTIONAL: '?';
+MISSING_EVENT_OPEN: '-[';
```

Comment:
I know we've discussed this over and over again, and I've re-read now everything around this subject. While I find it more logical and syntax appropriate to use a suffix for the missing events feature, in combination with the `with` construct (think of `with runs=2, missing=true`) there are too many drawback to this approach and one compelling advantage to a prefix alternative, to push `with missing` forward. As such, I think we should use a prefix (like you suggested in this PR). One detail (that you already implemented) is that we don't have to have whitespaces between the "missing" character and the opening square bracket.

Also, I tend to prefer `!`. Minus sign has too many other implications imo. So the syntax should be `sequence by k1 with maxspan=1s [ test5 where tag == \"normal\" ] by k2 with runs=2 ![ test5 where tag == \"missing\" ] by k2 [ test5 where tag == \"normal\" ] by k2`

Reply:
The minus was just a temporary decision that allowed us to proceed with the implementation without waiting for an agreement on the final syntax.
I'm absolutely fine with `![`, I'm doing the change now.